R/QueueDataWarehouse.R

```diff
@@ -1,164 +1,206 @@
+  report.description <- c()
+  report.description$reportDescription <- c(data.frame(matrix(ncol=0, nrow=1)))
+  report.description$reportDescription$dateFrom <- unbox(date.from)
+  report.description$reportDescription$dateTo <- unbox(date.to)
+  report.description$reportDescription$reportSuiteID <- unbox(reportsuite.id)
+  
+  # no date granularity handling; we need to remove this arg from description so
+  #  set to NULL and Filter out later
+  if(date.granularity == "" || is.na(date.granularity)) {
+    date.granularity <- NULL
+  }
+  report.description$reportDescription$dateGranularity <- unbox(date.granularity)
+  
+  #Hack in locale, every method calls ApiRequest so this hopefully works
+  #Set encoding to utf-8 as well; if someone wanted to do base64 they are out of luck
+  report.description$reportDescription$locale <- unbox(AdobeAnalytics$SC.Credentials$locale)
+  report.description$reportDescription$elementDataEncoding <- unbox("utf8")
+  
+  
+  #If segment is null, apply the standard segment unbox function
+  #report.description$reportDescription$segments <- unbox(segment.id)
+  
+  if(as.list(segment.id)[1]==''){
+    #report.description$reportDescription$segment_id <- unbox(segment.id)
+    report.description$reportDescription$segments <- unbox(segment.id)
+  }
+  #If segment is not null, treat it like a list of metrics.
+  else{
+    report.description$reportDescription$segments <- data.frame( id = segment.id)
+    
+  }
+  
+  if(expedite==TRUE) {
+    report.description$reportDescription$expedite <- unbox(expedite)
+  }
+  report.description$reportDescription$metrics = data.frame(id = metrics)
+  
+  report.description$reportDescription$metrics = data.frame(id = metrics)
+  
+  # bring over classifications; first instantiate the default df for elements
+  elements_df <- data.frame(
+    id = elements
+  )
+  # if supplied, classification must be equal in length to elements
+  # handle scenarios where classification is supplied blank string, NA, 
+  #  or a mix thereof
+  classification_clean <- as.character(classification)
+  classification_clean[is.na(classification_clean)] <- ""
+  
+  if(!all(classification_clean == "") || length(classification_clean) > 0L) {
+    if(length(classification_clean) != length(elements)) {
+      stop(
+        paste(
+          "When supplied, 'classification' must be a character vector equal", 
+          "in length to 'elements'"
+        )
+      )
+    }
+    elements_df[["classification"]] <- classification_clean
+  }
+  
+  report.description$reportDescription$elements <- elements_df
+  
+  report.description$reportDescription$source <- unbox("warehouse")
+  
+  if(enqueueOnly==TRUE){
+    report.description$reportDescription$ftp <- unbox(data.frame(ftp))
+  }
+  
+  # filter out NULL; currently only applicable if user does not want any
+  #  date granularity
+  report.description <- Filter(function(x) !is.null(x), report.description)

```

Comment:
Pretty sure setting a field equal to NULL removes it when the `toJSON` is called. So unless I'm incorrect, then this line is probably unnecessary

Reply:
Let me double check to be absolutely certain, and will address in next commit.